internal/pkg/cli/pipeline_update.go

```diff
@@ -197,6 +210,36 @@ func (o *updatePipelineOpts) deployPipeline(in *deploy.CreatePipelineInput) erro
 	return nil
 }
 
+func (o *updatePipelineOpts) ConnectionName() (string, error) {
+	if o.repoURL == "" {
+		return "", fmt.Errorf("unable to locate the repository URL")
+	}
```

Comment:
can we remove `repoURL` as a field since we can typecast to a `BitbucketSource`?

Right now this field is only set for bitbucket repos instead of all repositories which makes our code more bug-prone